### ISSUE TITLE
REQUIRED: Support for transpileOnly - not really an option

### ISSUE THREAD

**Issue by bradennapier** [samchon/typia#402]
## Question

@samchon 

First of all, the general idea and what you have done is great.  Do not get me wrong. However, this is not really useable in a realistic environment where it might be useful.

I may be doing something wrong here, of course.

- Dev Experience becomes terrible.  
- Can not use `transpileOnly` which means to test your code without building you need to compile the entire application multiple times for every change.  
- You can not test with type errors because of this... our project is a monorepo with hundreds of thousands of lines of code and it is causing a bunch of type errors i otherwise would not have ... unless I turn off skip lib check

```
[12/06 12:46:56:323 pm] | [ ERROR   ] | CRITICAL | uncaughtException 
  |>  src/core/models/userWalletsMap.ts:28:10 - error TS2564: Property 'userId' has no initializer and is not definitely assigned in the constructor.
  |   
  |   28   public userId: number;
```

I get around 25,000 errors in our project (which runs and compiles fine otherwise) when trying to make this work by turning off transpileOnly to run in a development environment.

This file is a Sequelize type -- the userId is supplemented at compile time by the `sequelize-typescript` decorator which makes it valid.  Since we skipLibCheck (otherwise it wont compile ever and will take 1,000x longer literally when it does), it fails at runtime 

Basically - unless you want to compile every change to test code -- this is a fairly useless package. 

I guess the only solution would be to isolate ALL use of this module to export factory functions and compile those... but then your ts files are importing javascript files which is another problem so not sure how that would work.

Basically instead of `halt` this package should detect transpile only and perhaps instead implement non-optimized code (JSON.stringify()) since we know at compile time it will work as expected

**Comment by samchon** [comment]
25,000 compilation errors are your problem, not of `typescript-json`. Of course, ignoring compile error of TypeScript compiler by using `transpileOnly` option, but I wanna ask you that "Why do you use TypeScript unless you do not run type checking?" If you hope to ignore compilation errors continuously, I just want to recommend you to use JavaScript instead.

Also, you told me that `compile the entire application multiple times for every change`, but you can avoid such entire source compilation by just utilizing watch mode compilation. With the watch mode compilation, only a little piece of code would be compiled whenever changed.

  - `npx ttsc --watch`
  - https://www.typescriptlang.org/docs/handbook/configuring-watch.html

**Comment by bradennapier** [comment]
You are not understanding the situation.  We use transpileOnly because it is the RIGHT thing to do.

The point is it is giving errors that do not exist in reality

The error I bring up above... that is not an error that occurs normally

I can compile just fine

```
yarn build:all
```

works just fine

I am actually not sure why all those errors pop up but it is likely because we can not use `compilerHost` and we can not use `transpileOnly` when we use this module because when those are off i get the errors when they are on i do not (for both COMPILE and for ts-node).

The only problem comes in when trying to run `ts-node` with a full compile (which gives no actual benefit over just building the file from scratch on every change).

- When developing I don't care about little type errors... i am just testing the changes... but when we try to commit a file we run a full build and type check and only allow it if it passses


this only happens when trying to use `ts-node` and compile at same time 

https://typestrong.org/ts-node/docs/options#transpileonly

they even directly mention to use transpileOnly during dev in docs

https://typestrong.org/ts-node/docs/performance

**Comment by bradennapier** [comment]
To be clear - there are zero type errors in our project ... unless trying to run things as this requires us to. 

I can compile and run with your module no issues... but it requires we undo all development environment optimizations in order to use it at all

**Comment by samchon** [comment]
I'm sorry but supporting `transpileOnly` option is totally impossible. 

Therefore, only two way left:

1. Change your dev environment to use JS file built by watch mode compilation
2. Give up using `typescript-json`
